numpy/core/src/umath/dispatching.c

```diff
@@ -712,6 +711,11 @@ promote_and_get_info_and_ufuncimpl(PyUFuncObject *ufunc,
             return NULL;
         }
         else if (info != NULL) {
+            /* Add result to the cache using the original types: */
+            if (PyArrayIdentityHash_SetItem(ufunc->_dispatch_cache,
```

Comment:
Steals a reference to `info`?

Reply:
Currently the cache works with borrowed references, and `info` is all borrowed here, so it doesn't steal (but `getitem` returns a borrowed reference).  May be too confusing to keep it forever, though.

Reply:
Or `info` is a borrowed reference?

Reply:
Yes, `info` is borrowed.

Reply:
Not a problem if the function returning `info` is documented as returning a borrowed reference